ext/curl/interface.c

```diff
@@ -956,6 +962,66 @@ static size_t curl_write_header(char *data, size_t size, size_t nmemb, void *ctx
 }
 /* }}} */
 
+#if LIBCURL_VERSION_NUM >= 0x075000 /* 7.80.0 */
+static int curl_prereqfunction(void *clientp, char *conn_primary_ip, char *conn_local_ip, int conn_primary_port, int conn_local_port)
+{
+	php_curl *ch = (php_curl *)clientp;
+	php_curl_callback *t = ch->handlers.prereq;
+
+	int rval = CURL_PREREQFUNC_ABORT; /* Disallow the request by default. */
+
+#if PHP_CURL_DEBUG
+	fprintf(stderr, "curl_prereqfunction() called\n");
+	fprintf(stderr, "clientp = %x, conn_primary_ip = %s, conn_local_ip = %s, conn_primary_port = %d, conn_local_port = %d\n", clientp, conn_primary_ip, conn_local_ip, c, onn_primary_port, conn_local_port);
+#endif
+
+	zval argv[5];
+	zval retval;
+	zend_result error;
+	zend_fcall_info fci;
+
+	GC_ADDREF(&ch->std);
+	ZVAL_OBJ(&argv[0], &ch->std);
+	ZVAL_STRING(&argv[1], conn_primary_ip);
+	ZVAL_STRING(&argv[2], conn_local_ip);
+	ZVAL_LONG(&argv[3], conn_primary_port);
+	ZVAL_LONG(&argv[4], conn_local_port);
+
+	fci.size = sizeof(fci);
+	ZVAL_COPY_VALUE(&fci.function_name, &t->func_name);
+	fci.object = NULL;
+	fci.retval = &retval;
+	fci.param_count = 5;
+	fci.params = argv;
+	fci.named_params = NULL;
+
+	ch->in_callback = 1;
+	error = zend_call_function(&fci, &t->fci_cache);
```

Comment:
You could just use ``zend_call_known_fcc()`` considering we already have the FCC.